etcd/src/main/java/org/jclouds/etcd/EtcdApiMetadata.java

```diff
@@ -0,0 +1,87 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.jclouds.etcd;
+
+import java.net.URI;
+import java.util.Properties;
+
+import org.jclouds.apis.ApiMetadata;
+import org.jclouds.etcd.config.EtcdHttpApiModule;
+import org.jclouds.rest.internal.BaseHttpApiMetadata;
+
+import com.google.auto.service.AutoService;
+import com.google.common.collect.ImmutableSet;
+import com.google.inject.Module;
+
+@AutoService(ApiMetadata.class)
+public class EtcdApiMetadata extends BaseHttpApiMetadata<EtcdApi> {
+
+   public static final String API_VERSION = "v2";
+   public static final String BUILD_VERSION = "2.1.1";
+
+   @Override
+   public Builder toBuilder() {
+      return new Builder().fromApiMetadata(this);
+   }
+
+   public EtcdApiMetadata() {
+      this(new Builder());
+   }
+
+   protected EtcdApiMetadata(Builder builder) {
+      super(builder);
+   }
+
+   public static Properties defaultProperties() {
+      Properties properties = BaseHttpApiMetadata.defaultProperties();
+      return properties;
+   }
+
+   public static class Builder extends BaseHttpApiMetadata.Builder<EtcdApi, Builder> {
+
+      protected Builder() {
+         super(EtcdApi.class);
+         id("etcd").
+         name("Etcd API").
+         identityName("username:password").
+         defaultIdentity("Optional Username").
+         defaultCredential("Optional Password").
```

Comment:
I'm not saying you have to use ones or the others. All four methods should be present:

* `identityName` and `credentialName` are mandatory for all ApiMetadata classes. They are just labels that define the format of the identity and credential so users know what to put there when creating the context. They're just labels to give a hint and allow, for example, tooling to prompt for the right thing.
* `defaultIdentity` and `defaultCredential` are default values to set for the identity and credential fields, if that makes sense, in case the user does not supply one of them. These are optional as in most apis/providers users are required both the identity and the credential.

In this case, as the credentials are not used, you have to set all four methods in your api metadata, in order to configure a default dummy credentials so the user does not have to provide them (which would be useless and confusing) when creating the context.